src/modules/Dropdown/Dropdown.js

```diff
@@ -650,7 +650,20 @@ export default class Dropdown extends Component {
 
     if (!search) return this.toggle(e)
     if (open) return
-    if (searchQuery.length >= minCharacters || minCharacters === 1) this.open(e)
+    if (searchQuery.length >= minCharacters || minCharacters === 1) {
+      this.open(e)
+      return
+    }
+    if (this.searchRef) this.searchRef.focus()
+  }
+
+  handleIconClick = e => {
+    debug('handleIconClick()', e)
+
+    _.invoke(this.props, 'onClick', e, this.props)
+    // prevent handleClick()
+    e.stopPropagation()
+    this.toggle(e)
```

Comment:
I don't want to split this to separate method, however I don't see any other way. The initial idea was to compare `e.target` in `handleClick` this the `Icon`'s ref, but the `Icon` component is stateless and doesn't have refs.